DependencyInjection/DoctrineExtension.php

```diff
@@ -739,20 +741,42 @@ protected function loadCacheDriver($driverName, $entityManagerName, array $drive
         $serviceId = null;
         $aliasId   = $this->getObjectManagerElementName(sprintf('%s_%s', $entityManagerName, $driverName));
 
+        if ($driverMap['type'] === null) {
+            $driverMap = [
+                'type' => 'pool',
+                'pool' => $this->getPoolNameForCacheDriver($driverName),
+            ];
+        }
+
+        $usesDeprecatedCacheType = true;
+
         switch ($driverMap['type']) {
             case 'service':
-                $serviceId = $driverMap['id'];
+                $usesDeprecatedCacheType = false;
+                $serviceId               = $driverMap['id'];
                 break;
 
             case 'pool':
-                $serviceId = $this->createPoolCacheDefinition($container, $driverMap['pool']);
+                $usesDeprecatedCacheType = false;
+                $serviceId               = $this->createPoolCacheDefinition($container, $driverMap['pool']);
                 break;
 
             case 'provider':
                 $serviceId = sprintf('doctrine_cache.providers.%s', $driverMap['cache_provider']);
                 break;
         }
 
+        if ($usesDeprecatedCacheType) {
```

Comment:
Couldn't this be done in the `Configuration` class?

Reply:
How so? I’d have to check the actual value given to the type configuration option and can’t deprecate the entire configuration option.

Reply:
Using a normalizer and calling `trigger_error()` should work, right?

Reply:
I'll give it a shot 👍